Lib/test/test_string.py

```diff
@@ -475,6 +475,55 @@ class PieDelims(Template):
         self.assertEqual(s.substitute(dict(who='tim', what='ham')),
                          'tim likes to eat a bag of ham worth $100')
 
+    def test_is_valid(self):
+        eq = self.assertEqual
+        s = Template('$who likes to eat a bag of ${what} worth $$100')
+        valid = s.is_valid()
+        eq(valid, True)
+
+        s = Template('$who likes to eat a bag of ${what} worth $100')
+        valid = s.is_valid()
+        eq(valid, False)
+
+        class BadPattern(Template):
+            pattern = r"""
+            (?P<badname>.*)                  |
+            (?P<escaped>@{2})                   |
+            @(?P<named>[_a-z][._a-z0-9]*)       |
+            @{(?P<braced>[_a-z][._a-z0-9]*)}    |
+            (?P<invalid>@)                      |
+            """
+        s = BadPattern('@bag.foo.who likes to eat a bag of @bag.what')
+        self.assertRaises(ValueError, s.is_valid)
```

Comment:
I wouldn't expect `is_valid()` to ever raise an exception.  Why wouldn't it just return `False` here?

Reply:
Resolving this as discussed above, however I do think the documentation should explain this behavior.